webapp/src/js/components/Map/TableMarkersLayer.js

```diff
@@ -82,6 +88,37 @@ let TableMarkersLayer = React.createClass({
     };
   },
 
+  componentDidMount() {
+    let {table, markerColourProperty} = this.props;
+    let {addControl} = this.context;
+
+    if (table !== undefined && markerColourProperty !== undefined && addControl !== undefined) {
+      let legendControl = this.composeLegendControl({table, markerColourProperty});
+      this.legendControlId = uid(10);
+      legendControl.id = this.legendControlId;
+      addControl(legendControl);
+    }
+  },
+
+  componentDidUpdate(prevProps, prevState) {
+    let {table, markerColourProperty} = this.props;
+    let {changeControl} = this.context;
+
+    if (table !== undefined && markerColourProperty !== undefined && changeControl !== undefined) {
+      let legendControl = this.composeLegendControl({table, markerColourProperty});
+      legendControl.id = this.legendControlId;
+      changeControl(legendControl);
+    }
+  },
+
+  componentWillUnmount() {
+    let {removeControl} = this.context;
+
+    if (removeControl !== undefined) {
```

Comment:
I expect that's a superfluous relic from when removeControl wasn't defined by Map's getChildContext(); I'll trim it out.